tests/Unit/Subscriptions/Iterators/GuardContextSyncIteratorTest.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Tests\Unit\Subscriptions\Iterators;
+
+use Exception;
+use GraphQL\Type\Definition\ResolveInfo;
+use Illuminate\Auth\AuthManager;
+use Illuminate\Contracts\Auth\Authenticatable;
+use Illuminate\Http\Request;
+use Illuminate\Support\Collection;
+use Mockery\MockInterface;
+use Nuwave\Lighthouse\Schema\Context;
+use Nuwave\Lighthouse\Subscriptions\Iterators\GuardContextSyncIterator;
+use Nuwave\Lighthouse\Subscriptions\Subscriber;
+use Nuwave\Lighthouse\Subscriptions\SubscriptionGuard;
+use Tests\Unit\Subscriptions\SubscriptionTestCase;
+
+class GuardContextSyncIteratorTest extends SubscriptionTestCase
+{
+    /**
+     * @var string
+     */
+    public const EXCEPTION_MESSAGE = 'test_exception';
+
+    /**
+     * @var \Nuwave\Lighthouse\Subscriptions\Iterators\GuardContextSyncIterator
+     */
+    protected $iterator;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->iterator = $this->app->make(GuardContextSyncIterator::class);
+    }
+
+    public function testCanIterateOverItemsWithCallback(): void
+    {
+        $items = [];
+
+        $this->iterator->process(
+            $this->subscribers(),
+            static function ($item) use (&$items): void {
+                $items[] = $item;
+            }
+        );
+
+        $this->assertCount(3, $items);
+    }
+
+    public function testCanPassExceptionToHandler(): void
+    {
+        /** @var \Exception|null $exception */
+        $exception = null;
+
+        $this->iterator->process(
+            $this->subscribers(),
+            static function (): void {
+                throw new Exception(self::EXCEPTION_MESSAGE);
+            },
+            static function (Exception $e) use (&$exception): void {
+                $exception = $e;
+            }
+        );
+
+        $this->assertSame(self::EXCEPTION_MESSAGE, $exception->getMessage());
```

Comment:
Nit: This does not validate the exact same exception is passed. You can store it in a variable before throwing and compare object equality.